incidents/2019-02-20--versionista-scraper-silent-failure.md

```diff
@@ -0,0 +1,84 @@
+
+NOTE: the PR to fix this ([scraper#196](https://github.com/edgi-govdata-archiving/web-monitoring-versionista-scraper/pull/196)) is unmerged and still awaiting review, but is running in production.
+
+
+## Timeline
+
+All times in PST.
+
+### 2019-02-20 14:01
+
+@jjudish posts [an issue on the versionista-scraper project](https://github.com/edgi-govdata-archiving/web-monitoring-versionista-scraper/issues/195) describing unusual error messages they haven’t seen before. (“No versions found for {URL}”)
+
+### 2019-02-20 17:47
+
+@Mr0grog responds with some possibilities and thoughts for debugging from his phone.
+
+### 2019-02-20 19:00
+
+@Mr0grog gets home and checks logs to see if we are having similar issues — we are. The issues started on Feb 12th at 3pm PST (`2019-02-12T23:00:00Z`). Checking the page in browsers demonstrates the issue immediately: the data is no longer being delivered in the HTML body of the page.
+
+### 2019-02-20 21:20
+
+@Mr0grog creates hotfix code to:
+
+1. Throw exceptions in this situation instead of treating it the same as a page with no versions.
+2. Throw exceptions in a similar situation for lists of pages in a site and list of sites in an account, since it seem only natural that they will change in the same way soon.
+3. Use the same API call the content of the page is making to extract the data.
+
+He switches production to the branch with the hotfix after testing with a smaller timeframe locally and tests with a larger timeframe matching the first failing request from a week ago.
+
+### 2019-02-20 23:20
+
+@Mr0grog creates PR from the hotfix code after it seems to be working successfully and asks for @jjudish’s review. He then starts incrementally backfilling data from the `versionista1` account.
+
+### 2019-02-21 01:40
+
+The backfilling process finishes for `versionista1`. @Mr0grog starts a much larger single backfill process (instead of incremental) for `versionista2`.
+
+### 2019-02-21 01:50
+
+The backfilling process throws a few errors and @Mr0grog adjusts the PR to account for them.
+
+### 2019-02-21 02:30
+
+The backfilling process for `versionista2` appears to be going smoothly and @Mr0grog and goes to sleep. (Sites in the `versionista2` account tend to change with greater frequency, so the same operation usually takes 1.5x - 2x the time for `versionista2`, so @Mr0grog expected this to go for a while.)
+
+### 2019-02-21 21:00
+
+Backfilling process for `versionista2` finishes.
+
+
+## Lessons
+
+### What Went Well
+
+- The issue filed by @jjudish was super useful.
+- Logs provided URLs and obvious spots to investigate.
+
+
+### What Went Wrong
+
+- This didn’t register as an actual error and we had no alerts about it for a week.
+- Other people saw the issue before @Mr0grog but did not respond or investigate or flag it for someone who could.
```

Comment:
Who (besides @jjudish) saw this before you? Was this an analyst/dev communication gap?

Reply:
I know at least @lightandluck did, because the reason I saw the issue as early as I did was his comment on Slack. But he did not note that there was a potential major data problem: https://archivers.slack.com/archives/C47DJK8H0/p1550710886000900

(That’s actually why the first entry in the timeline is me replying to the issue from my phone and noting I’d take a look when I got to a computer.)

Reply:
Yea, sorry, I really didn't understand how big of a problem it was! I assumed it was just a problem on their end because I thought we had things covered by Sentry. I didn't think troubleshooting someone else's instance would be urgent. But know now that if someone is having problems with our software, we might have the same problem! Will remember in the future.

(Really I was dumbstruck by the fact that someone else was using the scraper for a year and we had never heard anything from them. But finding out it was WIP was anti-climatic)

Reply:
Thanks for the context!

Reply:
Man, this makes me really want to go read all the other incident reports. Good lesson learned. Thanks y'all for your diligence 👍 💯 💪